fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/Ordering.java

```diff
@@ -650,16 +651,29 @@ public static ImmutableSetMultimap<Value, Binding> normalizeBindingMap(@Nonnull
     @Nonnull
     public static PartiallyOrderedSet<Value> normalizeOrderingSet(@Nonnull final SetMultimap<Value, Binding> bindingMap,
                                                                   @Nonnull final PartiallyOrderedSet<Value> orderingSet) {
-        final var transitiveClosure = orderingSet.getTransitiveClosure();
         final var normalizedDependencyMapBuilder = ImmutableSetMultimap.<Value, Value>builder();
-
-        for (final var dependency : transitiveClosure.entries()) {
-            Verify.verify(bindingMap.containsKey(dependency.getKey()));
-            Verify.verify(bindingMap.containsKey(dependency.getValue()));
-            final boolean isFixed = areAllBindingsFixed(bindingMap.get(dependency.getKey())) ||
-                    areAllBindingsFixed(bindingMap.get(dependency.getValue()));
-            if (!isFixed) {
-                normalizedDependencyMapBuilder.put(dependency);
+        final var dependencyMap = orderingSet.getDependencyMap().asMap();
```

Comment:
This is the only meat of the PR. Using the transitive closure is expensive, this way of doing it is much faster.